recipes/box2d/2.4.x/conanfile.py

```diff
@@ -15,56 +18,48 @@ class Box2dConan(ConanFile):
     options = {"shared": [True, False],
                "fPIC": [True, False]}
     default_options = {"shared": False, "fPIC": True,}
-
-    exports_sources = ["CMakeLists.txt", "patches/**"]
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    exports_sources = ["patches/**"]
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
 
```

Comment:
```suggestion

    def layout(self):
        cmake_layout(self, src_folder="src")

```
Using different source folder, avoids mixing both source files with cmake generated files